purchase_sale_inter_company/models/stock_picking.py

```diff
@@ -103,25 +96,32 @@ def _action_done_intercompany_actions(self, purchase):
             else:
                 self._notify_picking_problem(purchase)
 
-    def _notify_picking_problem(self, purchase):
+    def _notify_picking_problem(self, purchase, additional_note=False):
         self.ensure_one()
         note = _(
             "Failure to confirm picking for PO %s. "
             "Original picking %s still confirmed, please check "
             "the other side manually."
         ) % (purchase.name, self.name)
-        self.activity_schedule(
+        if additional_note:
+            note += _(" Additional info: ") + additional_note
+        user_id = self.sudo()._get_user_to_notify(purchase)
+        self.sudo().activity_schedule(
             "mail.mail_activity_data_warning",
             fields.Date.today(),
             note=note,
-            # Try to notify someone relevant
-            user_id=(
+            user_id=user_id or SUPERUSER_ID,
+        )
+
+    def _get_user_to_notify(self, purchase):
+        """Notify user based on res.config.settings"""
+        if self.company_id.notification_side == "so":
+            return (
                 self.company_id.notify_user_id.id
                 or self.sale_id.user_id.id
                 or self.sale_id.team_id.user_id.id
-                or SUPERUSER_ID,
-            ),
-        )
+            )
+        return purchase.user_id.id
```

Comment:
This should rather be `purchase.company_id.notify_user.id or purchase.user_id.id`, I think, so that it can be configured on the target company side.

Reply:
I see. So in res.config.settings, I have made notify_user_id a required field, only when notification_side is configured for source company sale.order. So first I should remove this restriction, and then make this change

Reply:
Hmm, yes.